Backpack/src/main/java/net/skyscanner/backpack/util/TextLayoutCompat.kt

```diff
@@ -0,0 +1,182 @@
+    textDir: TextDirectionHeuristic = TextDirectionHeuristics.FIRSTSTRONG_LTR,
+    spacingMult: Float = 1f,
+    spacingAdd: Float = 0f,
+    includePad: Boolean,
+    ellipsize: TextUtils.TruncateAt? = null,
+    ellipsizedWidth: Int = outerWidth,
+    maxLines: Int = Int.MAX_VALUE,
+  ): StaticLayout = creator.create(
+    source = source,
+    bufStart = bufStart,
+    bufEnd = bufEnd,
+    paint = paint,
+    outerWidth = outerWidth,
+    align = align,
+    textDir = textDir,
+    spacingMult = spacingMult,
+    spacingAdd = spacingAdd,
+    includePad = includePad,
+    ellipsize = ellipsize,
+    ellipsizedWidth = ellipsizedWidth,
+    maxLines = maxLines,
+  )
+
+  interface StaticLayoutCreator {
+
+    fun create(
+      source: CharSequence,
+      bufStart: Int,
+      bufEnd: Int,
+      paint: TextPaint,
+      outerWidth: Int,
+      align: Layout.Alignment,
+      textDir: TextDirectionHeuristic,
+      spacingMult: Float,
+      spacingAdd: Float,
+      includePad: Boolean,
+      ellipsize: TextUtils.TruncateAt?,
+      ellipsizedWidth: Int,
+      maxLines: Int
+    ): StaticLayout
+  }
+
+  object StaticLayoutCreatorApi21 : StaticLayoutCreator {
+
+    private val ConstructorReference: Constructor<StaticLayout>? = try {
+      StaticLayout::class.java.getDeclaredConstructor(
+        CharSequence::class.java,
+        Int::class.javaPrimitiveType,
+        Int::class.javaPrimitiveType,
+        TextPaint::class.java,
+        Int::class.javaPrimitiveType,
+        Layout.Alignment::class.java,
+        TextDirectionHeuristic::class.java,
+        Float::class.javaPrimitiveType,
+        Float::class.javaPrimitiveType,
+        Boolean::class.javaPrimitiveType,
+        TextUtils.TruncateAt::class.java,
+        Int::class.javaPrimitiveType,
+        Int::class.javaPrimitiveType,
+      ).apply {
+        isAccessible = true
+      }
+    } catch (t: Throwable) {
+
+      if (BuildConfig.DEBUG) {
+        Log.e(TAG, "Failed to find multi-line StaticLayout constructor using reflection", t)
+      }
+
+      null
+    }
+
+    @Suppress("DEPRECATION")
+    override fun create(
+      source: CharSequence,
+      bufStart: Int,
+      bufEnd: Int,
+      paint: TextPaint,
+      outerWidth: Int,
+      align: Layout.Alignment,
+      textDir: TextDirectionHeuristic,
+      spacingMult: Float,
+      spacingAdd: Float,
+      includePad: Boolean,
+      ellipsize: TextUtils.TruncateAt?,
+      ellipsizedWidth: Int,
+      maxLines: Int,
+    ): StaticLayout = try {
+
+      ConstructorReference?.newInstance(
+        source, bufStart, bufEnd, paint, outerWidth, align, textDir, spacingMult,
+        spacingAdd, includePad, ellipsize, ellipsizedWidth, maxLines,
+      ) ?: throw NullPointerException("StaticLayoutConstructor is null!")
+    } catch (t: Throwable) {
+
+      if (BuildConfig.DEBUG) {
+        Log.e(TAG, "Failed to create StaticLayout instance using reflection.", t)
+      }
+
+      StaticLayout(
```

Comment:
And return default StaticLayout created by public API. It won't be limited by 2 lines, but it will work in the same way. It's better than crashing the app, and this should never happen in real life.